### DISCUSSION TITLE
prisma command not found

### DISCUSSION THREAD

**Discussion by NickSchmitt** [prisma/prisma#4978] [Q&A] [answered]
I have tried following the [howtographql](https://www.howtographql.com/) tutorial and the [prisma quick starter](https://www.prisma.io/docs/getting-started/quickstart-typescript#write-your-first-query-with-prisma-client), and for both of them I am having problems with Prisma. Does this have to do with the upgrade from Prisma 1 to 2? I have tried various installation methods for Prisma and I am getting a `prisma: command not found` on all of them. What is the issue here?

on npm run dev I receive:

```

> dev
> ts-node ./script.ts


/Users/nock/Documents/Docs-Icloud/Development/GraphQL/GraphQL-Prisma-Starter/node_modules/.prisma/client/index.js:3
    throw new Error(
          ^
Error: @prisma/client did not initialize yet. Please run "prisma generate" and try to import it again.
In case this error is unexpected for you, please report it in https://github.com/prisma/prisma-client-js/issues/390.
```

**Comment by nikolasburk** [comment, marked as answer]
Hey @NickSchmitt, thanks for opening this discussion! 

Just to confirm, did you follow the _exact_ instructions in the [Quickstart](https://www.prisma.io/docs/getting-started/quickstart-typescript) guide? Running the following commands _should_ work:

```
curl -L https://pris.ly/quickstart | tar -xz --strip=2 quickstart-master/typescript/starter
cd starter
npm install
```

Now replace the empty `main` function in `script.ts` with this:

```js
async function main() {
  const allUsers = await prisma.user.findMany()
  console.log(allUsers)
}
````

And finally run:

```
npm run dev
```

This should print the following:

```
[
  { id: 1, email: "sarah@prisma.io", name: "Sarah" },
  { id: 2, email: "maria@prisma.io", name: "Maria" },
]
```

If you follow these steps and don't the above output, something indeed seems fishy 🤔  I am not able to reproduce an error when I follow these steps, so I'd be curious to hear if that also solves the problem for you! If not, I'm happy to dig a bit deeper with you! Maybe you can zip your project and share it with me so I can check what the problem might be?

**Reply by nikolasburk** [reply]
Also, if you've seen the `prisma: command not found` output it seems like your machine doesn't know that `prisma` is a command you can run in the terminal.

We recommend the following to ensure that your installation can be used. In your local npm project, install the `@prisma/cli` as a development dependency. 

```
npm install @prisma/cli --save-dev
```

Installing the CLI as a development dependency enables you to invoke the `prisma` command by prefixing it with `npx`. So you can now run the following to invoke the Prisma CLI:

```
npx prisma
```

Did you try this approach as well? We have more documentation around this topic [here](https://www.prisma.io/docs/concepts/components/prisma-cli/installation#local-installation-recommended).

**Reply by vadymchornyidev** [reply]
Hey! In my case, I fixed it via `npx prisma generate`.

**Reply by komali2** [reply]
IMO the instructions https://www.prisma.io/docs/orm/overview/introduction/what-is-prisma#generating-prisma-client should be updated to reflect this

**Reply by komali2** [reply]
Also to future search engine people:

```
  The package @prisma/cli has been renamed to prisma.                     │
npm ERR! │                                                                             │
npm ERR! │     Please uninstall @prisma/cli first.                                     │
npm ERR! │     Then install prisma to continue using Prisma CLI:                       │
npm ERR! │                                                                             │
npm ERR! │         # Uninstall old CLI                                                 │
npm ERR! │         npm uninstall @prisma/cli                                           │
npm ERR! │                                                                             │
npm ERR! │         # Install new CLI                                                   │
npm ERR! │         npm install prisma --save-dev                                       │
npm ERR! │                                                                             │
npm ERR! │         # Invoke via npx                                                    │
npm ERR! │         npx prisma --help                                                   │
npm ERR! │                                                                             │
npm ERR! │     Learn more here: https://github.com/prisma/prisma/releases/tag/2.16.0 
```

**Comment by NickSchmitt** [comment]
Hi @nikolasburk, thank you.

The error included `Please run "prisma generate" and try to import it again.`, but running prisma generate gave me `zsh: command not found: prisma`.  

However, `npx prisma generate` followed by `npm run dev` solved the problem.

**Reply by 0xVikasRushi** [reply]
Installing globally it works for me `npm i -g prisma `

**Comment by MR-ATESH** [comment]
`npx prisma [command]` worked very well for me

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
